samples/drivers/led_max7219/src/main.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <drivers/spi.h>
+
+#define GPIO_DEV "GPIO_0"
+
+#if (CONFIG_SPI_0 - 0) && defined(DT_SPI_0_NAME)
```

Comment:
While I think this sample is nice to see, I think @anangl is right about the need to extract this into a proper driver if this is going to be ready to be merged into mainline Zephyr. This has to do with the ability to make main.c truly independent of the board as requested in my initial review. This update is not quite enough, I am afraid.

What you'd need to do is write device tree bindings for the max7219 based on spi-device.yaml (search for examples in dts using `git grep`; the file dts/bindings/display/ilitek,ili9340.yaml looks like it might be a good starting point since the max7219 also looks like a display to me, but you might also look at the led_strip API and dts/bindings/led_strip/colorway,lpd8806.yaml).

Then you can write a driver using these bindings.

Finally, you can change this sample around to use board-specific device tree overlays, kind of like how samples/drivers/led_lpd8806/96b_carbon.overlay selects the spi device for that board.

Reply:
Ok, will do. But since this is serious scope creep, it will take some time to deliver.